src/api/main.py

```diff
@@ -76,7 +76,28 @@ async def lifespan(app: fastapi.FastAPI):
 
         app.state.ai_client = ai_client
         app.state.agent = agent
-
+        
+        
+        file_map: Dict[str, Dict[str, str]] = {}
+
+        logger.info(f"Creating UPLOADED_FILE_MAP")
+        if agent.tool_resources and agent.tool_resources.file_search:
+            for vector_store_id in agent.tool_resources.file_search.vector_store_ids:
+                file = await ai_client.agents.list_vector_store_files(vector_store_id)
+                for file in file.data:
```

Comment:
Technically, it will work for our case, but if the number of files will be more then `limit` (default 20), we may get issue. We may add additional cycle around:
```python
has_more = True
after_id = None
while has_more:
    file = await ai_client.agents.list_vector_store_files(vector_store_id, after=after_id)
    after_id = file.last_id
    has_more  = file.has_more
    <The internal cycle>
```